### PR TITLE
defn: Ordered families of equivalences

### PR DESCRIPTION
**Pull request by plt-amy**
OFEs are a presentation of certain metric spaces which see some use in the literature on guarded recursion/step indexing, e.g. the “standard” model of the Iris logic lives in the category of COFEs.

### CODE REVIEW

File: src/Cat/Instances/OFE.lagda.md [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+
+    ≈-refl  : ∀ n {x} → within n x x
+    ≈-sym   : ∀ n {x y} → within n x y → within n y x
+    ≈-trans : ∀ n {x y z} → within n y z → within n x y → within n x z
+
+    bounded : ∀ x y → within 0 x y
+    step    : ∀ n x y → within (suc n) x y → within n x y
+    limit   : ∀ x y → (∀ n → within n x y) → x ≡ y
+
+  ofe→ids : is-identity-system (λ x y → ∀ n → within n x y) (λ x n → ≈-refl n)
```
Comment (TOTBWF) [inline src/Cat/Instances/OFE.lagda.md:61]: Very nice

File: src/Cat/Instances/OFE/Coproduct.lagda.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+We now prove that this construction, which despite my best efforts may
+still feel unmotivated, _is_ the categorical product in OFEs. We start
+by defining the coproduct of morphisms, the operation taking $f : A \to
+Q$ and $g : B \to Q$ to $[f,g] : A \uplus B \to Q$. Once again, it's a
+case bash, and we have to use boundedness of the codomain when showing
+that points with distance 1 stay with distance 1.
+
+```agda
+OFE-Coproduct : ∀ {ℓ ℓ'} A B → Coproduct (OFEs ℓ ℓ') A B
```
Comment (TOTBWF) [inline src/Cat/Instances/OFE/Coproduct.lagda.md:150]: Maybe break this up into `⊎-OFE-is-coproduct`, or at least give the `where` block a module?

File: src/Cat/Instances/OFE/Later.lagda.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    module P = OFE-on P
+```
+-->
+
+We can put this together to define something akin to **Löb induction**:
+if $A$ is an inhabited, [complete OFE], Banach's fixed point theorem
+implies that non-expansive functions $\later P \to P$ have fixed points.
+
+```agda
+  löb : ∥ A ∥ → is-cofe P → (f : Later P →ⁿᵉ P) → Σ A λ x → f .map x ≡ x
```
Comment (TOTBWF) [inline src/Cat/Instances/OFE/Later.lagda.md:127]: This is very slick

**Review by TOTBWF — DISMISSED** [review]

File: src/Cat/Instances/OFE/Coproduct.lagda.md (hunk 2 above)
Comment (plt-amy) [reply src/Cat/Instances/OFE/Coproduct.lagda.md:150]: Why?

File: src/Cat/Instances/OFE.lagda.md [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+    module O = OFE-on O
+    module P = OFE-on P
+```
+-->
+
+```agda
+  record is-contractive : Type (ℓ ⊔ ℓ' ⊔ ℓ₁') where
+    no-eta-equality
+    field
+      contract : ∀ {x y n} → O.within n x y → P.within (suc n) (f x) (f y)
```
[the remark this answers is not in the document]
Comment (plt-amy) [reply src/Cat/Instances/OFE.lagda.md:339]: yes, that equivalence is in the Later file

**Pull request merged**